### PR TITLE
Hnsecure message validation

### PR DESCRIPTION
**Pull request by AnumehaSrivastava05**

### CODE REVIEW

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/HL7Message.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
+package ca.bc.gov.hlth.hnsecure.message;
+
+public class HL7Message {
+
+	public String segmentIdentifier = "MSH";
```
Comment (tschia) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/HL7Message.java:5]: Should this be private and just use getters/setters like the rest of the variables

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java [hunk 2]
```diff
@@ -55,6 +61,8 @@ public void configure() {
             
             //dispatch the message based on the receiving application code and message type
             .choice()
+            	.when(simple("${body} contains 'AR'"))
+            	.log("There is an error validation the message and it will not be sent to endpoint")
```
Comment (tschia) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java:65]: We should implement the code to handle these validation errors as part of this story. 

I think the best way to do this is to throw a custom exception from the Validator and to have an `onException().handled(true)` handler in the Route.java that catches the validation exception.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/messagevalidation/V2PayloadValidator.java [hunk 3]
```diff
@@ -1,45 +1,179 @@
...
+			}
+		} else {
+			isValidSeg = false;
+			errorMessage = ErrorMessage.HL7Error_Msg_NoInputHL7;
+
+		}
+
+		// Validate segment identifier
+		if (isValidSeg && !messageObj.getSegmentIdentifier().equals(segmentIdentifier)) {
+			isValidSeg = false;
```
Comment (tschia) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/messagevalidation/V2PayloadValidator.java:80]: I think when a validation fails we want to just update the exchange body with the new response and throw an exception that can be handled in the route, something like: 

```
if (some validation fails) {
    exchange.setBody(createErrorResponse(errorMessage.SOME_ERROR_MSG));
    throw new ValidationFailedException();
}
```

**Comment by AnumehaSrivastava05** [comment]
Hi Trevor, Please review the changes done for error codes

**Comment by tschia** [comment]
The changes look good. I added a small refactor to the V2PayloadValidator just to make the conditional statements a little more readable.
 None of the actual validation logic should have changed. If it still looks good to you we can Merge this pull request.

**Pull request merged**